core/frontend/src/store/mavlink.ts

```diff
@@ -0,0 +1,64 @@
+import Vue from 'vue'
+import {
+  Action, Module, Mutation, VuexModule,
+} from 'vuex-module-decorators'
+
+import mavlink2rest from '@/libs/MAVLink2Rest'
+import Listener from '@/libs/MAVLink2Rest/Listener'
+import store from '@/store'
+import { Dictionary } from '@/types/common'
+import { MavlinkMessage } from '@/types/mavlink'
+
+interface messsageRefreshRate {
+  message: string
+  refreshRate: number
+}
+
+@Module({
+  dynamic: true,
+  store,
+  name: 'mavlink_store',
+})
+
+export default class MavlinkStore extends VuexModule {
+  available_messages: Dictionary<MavlinkMessage> = {}
+
+  message_listeners: Dictionary<Listener> = {}
+
+  @Action({ commit: 'updateMessage' })
+  setMessageRefreshRate(rate: messsageRefreshRate): void {
+    const { messageName, refreshRate } = rate
+    if (refreshRate < 0) {
+      console.warn(`Invalid request rate requested for message ${messageName}@${refreshRate}Hz`)
+    }
+
+    mavlink2rest.requestMessageRate(messageName, refreshRate)
+    // Remove any listener that has a lower frequency than requested
+    if (messageName in this.message_listeners) {
+      const currentRate = this.message_listeners[messageName].frequency
+      if (currentRate > refreshRate) {
+        console.warn(
+          `Request with higher rate already registered for message ${messageName}@${currentRate}Hz vs ${refreshRate}Hz`,
+        )
+        return
+      }
+      this.message_listeners[messageName].discard()
+    }
+
+    // Create a new listener
+    this.message_listeners[messageName] = mavlink2rest.startListening(messageName).setCallback((receivedMessage) => {
+      this.updateMessage({
+        messageName,
+        messageData: receivedMessage,
+        requestedMessageRate: refreshRate,
+      })
+    }).setFrequency(refreshRate)
+  }
+
+  @Mutation
+  updateMessage(message: MavlinkMessage): void {
+    if (message) {
+      Vue.set(this.available_messages, message.messageName, message)
```

Comment:
I don't know this `Vue.set` function, but I assume this is similar to `Object.assign`, right? If so, the later would be a more common choice, easier to read.

Reply:
I believe that doing with Vue.set will register and update the change of the value to update any other variable that needs to be updated in an vue event.

Reply:
```
Vue does not allow dynamically adding new root-level reactive properties to an already created instance. However, it’s possible to add reactive properties to a nested object using the Vue.set(object, propertyName, value) method:
```

Reply:
I was unable to find something related to it in Vue3

Reply:
I have a feeling that Vue.set() will be a lot more efficient here. This function has the potential to be called at a pretty high rate and to hold quite a bit of messages.
It may also trigger updates in non-changed data if we update the whole dict.

For the readability, a comment would make it clear.

Reply:
Maybe. Really don't know about performance comparisons. It sounds kind of abusing the method though (what is the relationship between the available_messages vuex state and Vue?).

Although, I don't think we need to block this PR because of this.

Reply:
I'll add a todo about it.